library/src/scala/tasty/util/ShowSourceCode.scala

```diff
@@ -654,11 +654,20 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
 
     def printTypeTree(tree: TypeTree): Buffer = tree match {
       case TypeTree.Synthetic() =>
-        printType(tree.tpe)
-        tree.tpe match {
-          case tpe @ Type.TypeRef(name, _) if name.endsWith("$") => this += ".type"
-          case tpe => this
-        }
+        def printTypeAndAnnots(tpe: Type): Buffer = tpe match {
+          case Type.AnnotatedType(tp, annot) =>
+            printTypeAndAnnots(tp)
+            this += " "
+            printAnnotation(annot)
+          case tpe @ Type.TypeRef(name, _) if name.endsWith("$") =>
+            printType(tpe)
+            this += ".type"
+          case tpe @ Type.SymRef(ClassDef(name, _, _, _, _), _) if name.endsWith("$") =>
+            printType(tpe)
+            this += ".type"
+          case tpe => printType(tpe)
+        }
+        printTypeAndAnnots(tree.tpe)
```

Comment:
`tests/run-with-compiler/i3947c.scala` failed due to the change, I cannot figure out why.

Reply:
It was because `scala.runtime.Null$` and `scala.runtime.Nothing$` where handled as modules. Those are now special cases.